## Simulations on a Roulette Wheel


|Bets|Payout|Probability|
|----|------|-----------|
|Odd|1:1|48.6%|
|Even|1:1|48.6%|
|Black|1:1|48.6%|
|Red|1:1|48.6%|
|1-18|1:1|48.6%|
|19-36|1:1|48.6%|

In [1]:
import random

bankroll = 100
stake = 1
count = 0
while bankroll>0:
    if random.uniform(0,1)<= .486:
        bankroll += stake
        stake = 1
    else:
        bankroll -= stake
        stake = 2*stake
    count +=1 

print(count) 

1795


In [2]:
br = [10,100,1000,10000,10000,1000000,1000000000]

def gamble(money, probability, stk):
    stake = stk
    count = 0
    br = money
    while money>0:
        if random.uniform(0,1)<= probability:
            br += stake
            stake = 1
        else:
            money -= stake
            stake = 2*stake
        count +=1 
    print(count)
    
for i in br:
    gamble(i, .486, 1)

19
57
344
2719
2076
244624
98627537


## Conclusions

No matter how much money you throw at this strategy you always lose. 

|Starting Amount|Cycles Untill Broke|
|-------|-------:|
|10|19|
|100|57|
|1,000|344|
|10,000|2,719|
|100,000|2,076|
|1,000,000|244,624|
|1,000,000,000|98,627,537|